upload.sh

```diff
@@ -1,44 +1,101 @@
 #!/bin/sh
 
-# Get tag from environment
+# Get release tag from environment and checkout branch.
 tag=$(basename $GITHUB_REF)
+git checkout tags/$tag
 
-# Get action that triggered release event
+# Get package version.
+version=$(python setup.py --version)
+
+# Check if release tag matches the package version.
+pip install --quiet packaging==19.2
+
+match=$(python -c "
+from packaging.version import parse
+
+match = parse('$tag') == parse('$version')
+print(match)
+")
+
+if [ $match = "False" ]; then
```

Comment:
The logic here is currently "do not release if $match evaluates to False" but I think better logic would be "only release if $match evaluates to True"

That way if something goes wrong and match is neither "True" or "False" we stop the action

Reply:
Sounds good. I updated the match logic.